opencti-platform/opencti-front/src/private/components/techniques/data_components/DataComponentDataSources.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { FunctionComponent } from 'react';
+import { graphql, useFragment, useMutation } from 'react-relay';
+import Typography from '@mui/material/Typography';
+import List from '@mui/material/List';
+import ListItem from '@mui/material/ListItem/ListItem';
+import { Link } from 'react-router-dom';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import { ProgressWrench } from 'mdi-material-ui';
+import ListItemText from '@mui/material/ListItemText';
+import { ListItemSecondaryAction } from '@mui/material';
+import IconButton from '@mui/material/IconButton';
+import { LinkOff } from '@mui/icons-material';
+import { useFormatter } from '../../../../components/i18n';
+import AddDataSources from './AddDataSources';
+import { DataComponentDataSources_dataComponent$data, DataComponentDataSources_dataComponent$key } from './__generated__/DataComponentDataSources_dataComponent.graphql';
+
+const dataComponentDataSourcesRemoveMutation = graphql`
+  mutation DataComponentDataSourcesRemoveMutation(
+    $id: ID!
+    $input: [EditInput]!
+  ) {
+    dataComponentFieldPatch(
+      id: $id
+      input: $input
+    ) {
+      ...DataComponentEditionOverview_dataComponent
+      ...DataComponent_dataComponent
+    }
+  }
+`;
+
+const DataComponentDataSourceFragment = graphql`
+  fragment DataComponentDataSources_dataComponent on DataComponent {
+    id
+    dataSource {
+      id
+      name
+      description
+    }
+  }
+`;
+
+interface DataComponentDataSourcesProps {
+  dataComponent: DataComponentDataSources_dataComponent$key
+}
+
+const DataComponentDataSources: FunctionComponent<DataComponentDataSourcesProps> = ({ dataComponent }) => {
+  const { t } = useFormatter();
+
+  const data: DataComponentDataSources_dataComponent$data = useFragment(DataComponentDataSourceFragment, dataComponent);
+
+  const dataSourceId: string | undefined = data.dataSource?.id;
```

Comment:
The component name suggest that we are looking at the datasourceS of a datacomponent, how can the query gives us a dataSource ID ? And how can it be undefined if we want it ?

Reply:
The name has been changed.
For the 'undefined' type, we are not sure if we have a datasource attached to the datacomponent (and we show the add button depending on that)